### ISSUE TITLE
Build Issue

### ISSUE THREAD

**Issue by mahee522** [cincheo/jsweet#407]
I trying with jsweet quickstart project POM is giving build error.....

POM.xml

`<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>org.jsweet</groupId>
	<artifactId>jsweet-quickstart</artifactId>
	<version>2.0.0</version>
	<name>JSweet v2 quick start</name>
	<description>A simple project structure to get started with JSweet v2</description>
	<developers>
		<developer>
			<id>rpawlak</id>
			<name>Renaud Pawlak</name>
			<email>renaud.pawlak@gmail.com</email>
		</developer>
</developers>
	<licenses>
		<license>
			<name>The Apache Software License, Version 2.0</name>
			<url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
			<distribution>repo</distribution>
		</license>
	</licenses>
	<pluginRepositories>
		<pluginRepository>
			<id>jsweet-plugins-release</id>
			<name>plugins-release</name>
			<url>http://repository.jsweet.org/artifactory/plugins-release-local</url>
		</pluginRepository>
		<pluginRepository>
			<snapshots />
			<id>jsweet-plugins-snapshots</id>
			<name>plugins-snapshot</name>
			<url>http://repository.jsweet.org/artifactory/plugins-snapshot-local</url>
		</pluginRepository>
	</pluginRepositories><repositories>
		<repository>
			<id>jsweet-central</id>
			<name>libs-release</name>
			<url>http://repository.jsweet.org/artifactory/libs-release-local</url>
		</repository>
		<repository>
			<snapshots />
			<id>jsweet-snapshots</id>
			<name>libs-snapshot</name>
			<url>http://repository.jsweet.org/artifactory/libs-snapshot-local</url>
		</repository>
	</repositories><build>
		<plugins>
			<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>3.1</version>
				<configuration>
					<source>1.8</source>
					<target>1.8</target>
					<fork>true</fork>
				</configuration>
			</plugin>
			<plugin>
				<groupId>org.jsweet</groupId>
				<artifactId>jsweet-maven-plugin</artifactId>
				<version>2.0.0</version>
				<configuration>
					<verbose>true</verbose>
					<tsOut>target/ts</tsOut>
					<outDir>target/js</outDir>
					<candiesJsOut>webapp</candiesJsOut>
					<targetVersion>ES3</targetVersion>
				</configuration>
				<executions>
					<execution>
						<id>generate-js</id>
						<phase>generate-sources</phase>
						<goals>
							<goal>jsweet</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins><pluginManagement>
			<plugins><plugin>
					<groupId>org.eclipse.m2e</groupId>
					<artifactId>lifecycle-mapping</artifactId>
					<version>1.0.0</version>
					<configuration>
						<lifecycleMappingMetadata>
							<pluginExecutions>
								<pluginExecution>
									<pluginExecutionFilter>
										<groupId>
											org.jsweet
										</groupId>
										<artifactId>
											jsweet-maven-plugin
										</artifactId>
										<versionRange>
											[1.0.0,)
										</versionRange>
										<goals>
											<goal>jsweet</goal>
										</goals>
									</pluginExecutionFilter>
									<action>
										<ignore></ignore>
									</action>
								</pluginExecution>
							</pluginExecutions>
						</lifecycleMappingMetadata>
					</configuration>
				</plugin>
			</plugins>
		</pluginManagement>
	</build>
	<dependencies>
		<dependency>
			<groupId>org.jsweet</groupId>
			<artifactId>jsweet-core</artifactId>
			<version>5-20170726</version>
		</dependency><dependency>
			<groupId>org.jsweet.candies</groupId>
			<artifactId>jquery</artifactId>
			<version>1.10.0-20170726</version>
		</dependency>
	</dependencies>
</project>`

**ISSUE:**
Failed to execute goal org.jsweet:jsweet-maven-plugin:2.0.0:jsweet (generate-js) on project jsweet-quickstart: Execution generate-js of goal org.jsweet:jsweet-maven-plugin:2.0.0:jsweet failed: Plugin org.jsweet:jsweet-maven-plugin:2.0.0 or one of its dependencies could not be resolved: Failed to collect dependencies at org.jsweet:jsweet-maven-plugin:jar:2.0.0 -> org.apache.maven:maven-plugin-api:jar:3.3.3: Failed to read artifact descriptor for org.apache.maven:maven-plugin-api:jar:3.3.3: Could not transfer artifact org.apache.maven:maven-plugin-api:pom:3.3.3 from/to jsweet-plugins-snapshots (http://repository.jsweet.org/artifactory/plugins-snapshot-local): Failed to transfer file: http://repository.jsweet.org/artifactory/plugins-snapshot-local/org/apache/maven/maven-plugin-api/3.3.3/maven-plugin-api-3.3.3.pom. Return code is: 409 , ReasonPhrase:Conflict. -> [Help 1]

**Comment by renaudpawlak** [comment]
Looks like a Maven issue. Can you try to clean up your local maven repo (.m2 directory) and try again?

**Comment by jsteinbrunner** [comment]
@renaudpawlak / @lgrignon 
I know this is an old issue, but I am running into something similar that I believe may be related. I'm new to JSweet (love what I'm seeing so far!), and trying to get a project setup.

`[ERROR] Failed to execute goal org.jsweet:jsweet-maven-plugin:3.0.0:jsweet (generate-js) on project jsweet-quickstart: Execution generate-js of goal org.jsweet:jsweet-maven-plugin:3.0.0:jsweet failed: Plugin org.jsweet:jsweet-maven-plugin:3.0.0 or one of its dependencies could not be resolved: Failed to collect dependencies at org.jsweet:jsweet-maven-plugin:jar:3.0.0 -> org.jsweet:jsweet-transpiler:jar:3.0.0 -> org.jsweet.ext:typescript.java-ts.core:jar:2.0.3: Failed to read artifact descriptor for org.jsweet.ext:typescript.java-ts.core:jar:2.0.3: Could not transfer artifact org.jsweet.ext:typescript.java-ts.core:pom:2.0.3 from/to jsweet-plugins-snapshots (http://repository.jsweet.org/artifactory/plugins-snapshot-local): Failed to transfer file: http://repository.jsweet.org/artifactory/plugins-snapshot-local/org/jsweet/ext/typescript.java-ts.core/2.0.3/typescript.java-ts.core-2.0.3.pom. Return code is: 409 , ReasonPhrase:Conflict.`

We are using our own artifactory repository that proxies to other repositories when artifacts are not found locally. In our maven settings, we have the respository URL set to `../plugins-release`.

I'm not a Maven expert, so I apologize if this is a dump question, but with this URL it is picking up the core libraries in the following path.
https://repository.jsweet.org/artifactory/webapp/#/artifacts/browse/tree/General/plugins-release/org/jsweet/jsweet-core

Should the core libraries live under `plugins-release`?

Thanks guys!

**Comment by lgrignon** [comment]
Could you please try with JSweet 3.1.0?

**Comment by jsteinbrunner** [comment]
Thanks for the reply @lgrignon !

I wasn't able to fix the problem by updating to `3.1.0`. However, I didn't realize the significance of the `<pluginRepository>` element in the pom.

From what I have read, maven uses this repository for plugin dependencies, so it makes sense to me now why those core libraries were in the plugin tree in artifactory. 

I was able to fix my problem by adding the release as a plugin repository.
```
    <pluginRepositories>
        <pluginRepository>
            <id>jsweet-plugins-release</id>
            <name>plugins-release</name>
            <url>http://repository.jsweet.org/artifactory/plugins-release-local</url>
        </pluginRepository>
        <pluginRepository>
            <id>jsweet-central</id>
            <name>libs-release</name>
            <url>http://repository.jsweet.org/artifactory/libs-release-local</url>
        </pluginRepository>
    </pluginRepositories>
```

I wonder though, should `org.jsweet.ext:typescript.java-ts.core:jar:2.0.3` and `org.jsweet.ext:sourcemap-builder:jar:1.0.0` also be in the plugin tree since they are dependencies of the plugin? Probably not a big deal, just thinking out loud mostly.

**Comment by lgrignon** [comment]
Not sure about your last question, I am not very surprised about it though :) 

I close this issue because it seems solved, please feel free to reopen if I'm missing something
